src/dataService/pocketSavesService.ts

```diff
@@ -137,11 +147,75 @@ export class PocketSaveDataService {
    */
   public async getListRowById(itemId: string): Promise<ListResult> {
     const query = await this.buildQuery()
-      .where({ user_id: this.userId, item_id: itemId })
+      .where('user_id', this.userId)
+      .andWhere('item_id', itemId)
       .first();
+    return PocketSaveDataService.convertListResult(query);
+  }
 
-    const rawResp = query;
-    const resp = PocketSaveDataService.convertListResult(rawResp);
-    return resp;
+  /**
+   * Batch update to set status of saves in a user's list to ARCHIVED.
+   * Requires all IDs in the batch to be valid; otherwise will roll back
+   * transaction and return missing IDs, for use in NOT_FOUND response
+   * in business layer.
+   * If the row was already ARCHIVED status, the method is a "no-op"
+   * (e.g. does not reset the time_read, time_updated, or api_id_updated values)
+   */
+  public async archiveListRow(
+    ids: number[],
+    timestamp: Date
+  ): Promise<{ updated: ListResult[]; missing: string[] }> {
+    const timeUpdate = mysqlTimeString(timestamp, config.database.tz);
```

Comment:
Yes, the timestamp is provided by clients rather than automatically because of offline syncing. Mobile clients can store changes locally and then sync with the server later. Not making it optional helps prevent mistakes on the consumer side, although online clients like Web Client could just use the time the server received the request.

Reply:
got it, thanks